### PR TITLE
derive FileConfig

### PR DESCRIPTION
**Pull request by ModProg**

### CODE REVIEW

File: crates/bonsaidb-macros/src/lib.rs
Comment (ecton) [inline crates/bonsaidb-macros/src/lib.rs:806]:
```suggestion
```

File: crates/bonsaidb-macros/src/lib.rs
Comment (ecton) [inline crates/bonsaidb-macros/src/lib.rs:813]: Huh, invoking `Qualified::new` works here? I think some of the other code has a more complex `as` expression that could be simplified. That doesn't need to be done in this PR, just so astounded that this works that I'm leaving the comment :)

Neat!

**Review by ecton — APPROVED** [review]
Looks great!

Due to some issues in main, I'm going to hold off on this merge until I've merged #250.

File: crates/bonsaidb-macros/src/lib.rs
Comment (ModProg) [reply crates/bonsaidb-macros/src/lib.rs:806]: Ups, must have forgotten to remove that when introducing the match :D

**Pull request merged**
